ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,130 @@ const base = url.substring(0, url.lastIndexOf('/'));
+
+      expect(isElementVisible(layout.$.error)).to.be.false;
+      expect(layout.element.tagName).to.equal('DUMMY-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('span').innerText).to.equal('dummy');
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+    });
+
+    test('Should not map model into layout when model is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+    });
+
+    test('Should not display the layout when href changes and it is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html"></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+      expect(layout.element.tagName).to.equal('DUMMY-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+
+      layout.href = '';
+      await flush();
+
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Validation', () => {
+    test('Should validate the layout when fields are valid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "valid", "data": "foo"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      sinon.spy(layout.element, 'validate');
+      sinon.spy(layout, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.true;
+      expect(nuxeoInput.invalid).to.be.false;
+      expect(nuxeoInput.validate.calledOnce).to.be.true;
+      expect(layout.element.validate.calledOnce).to.be.true;
+      expect(layout.validate.calledOnce).to.be.true;
+    });
+
+    test('Should invalidate the layout when native validation is invalid', async () => {
```

Comment:
Still not sure about this test purpose. What are we trying to validate?
`Should not run custom validation when native validation fails`? Because I believe we are testing both here and it may be more explicit.

Reply:
Yes, I guess the nature of the test changed with our iterations and that would describe better what we are expecting from the test.